plugin/keyring_vault/i_vault_curl.h

```diff
@@ -0,0 +1,23 @@
+#ifndef MYSQL_I_VAULT_CURL
+#define MYSQL_I_VAULT_CURL
+
+#include "i_keyring_key.h"
+#include "vault_credentials.h"
+
+namespace keyring {
+
+class IVault_curl : public Keyring_alloc
+{
+public:
+  virtual my_bool init(Vault_credentials *vault_credentials) = 0;
+  virtual my_bool list_keys(std::string *response) = 0;
```

Comment:
response is changed in this function so it cannot be const reference and according to google code guideline non-const references are forbidden (which I am agreeing with). Also to explain myself why I was not using references at all - they are forbidden at all in Oracle MySQL.